usecases/ratelimiter/limiter.go

```diff
@@ -0,0 +1,59 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package ratelimiter
+
+import "sync"
+
+// Limiter is a thread-safe counter that can be used for rate-limiting requests
+type Limiter struct {
+	lock    sync.Mutex
+	max     int
```

Comment:
is 0 valid value? does it mean no requests are allowed?
or <= 0 should be unlimited?

Reply:
Good point, I'll adjust it. 0 doesn't make sense, we can treat it as unlimited, as well.